contribs/src/main/scala/io/delta/storage/BaseExternalLogStore.scala

```diff
@@ -0,0 +1,283 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.storage
+
+import java.io.FileNotFoundException
+import java.net.URI
+import java.nio.charset.StandardCharsets.UTF_8
+import java.nio.file.FileSystemException
+
+import com.google.common.io.CountingOutputStream
+import org.apache.commons.io.IOUtils
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs._
+import org.apache.spark.SparkConf
+import org.apache.spark.sql.delta.metering.DeltaLogging
+import org.apache.spark.sql.delta.util.FileNames
+import org.apache.spark.sql.delta.storage.HadoopFileSystemLogStore
+
+abstract class BaseExternalLogStore(sparkConf: SparkConf, hadoopConf: Configuration)
+  extends HadoopFileSystemLogStore(sparkConf, hadoopConf)
+    with DeltaLogging {
+
+  /**
+   * Delete file from filesystem.
+   * @param fs reference to [[FileSystem]]
+   * @param path path to delete
+   * @return Boolean true if delete is successful else false
+   */
+  private def deleteFile(fs: FileSystem, path: Path): Boolean = {
+    logDebug(s"delete file: $path")
+    fs.delete(path, false)
+  }
+
+  /**
+   * Copies file within filesystem
+   * @param fs reference to [[FileSystem]]
+   * @param src path to source file
+   * @param dst path to destination file
+   */
+  private def copyFile(fs: FileSystem, src: Path, dst: Path) {
+    logDebug(s"copy file: $src -> $dst")
+    val input_stream = fs.open(src)
+    val output_stream = fs.create(dst, true)
+    try {
+      IOUtils.copy(input_stream, output_stream)
+    } finally {
+      output_stream.close()
+      input_stream.close()
+    }
+  }
+
+  /**
+   * Check if the path is an initial version of a Delta log.
+   */
+  private def isInitialVersion(path: Path): Boolean = {
+    FileNames.isDeltaFile(path) && FileNames.deltaVersion(path) == 0L
+  }
+
+  /**
+   * Merge two iterators of [[FileStatus]] into a single iterator ordered by file path name.
+   * In case both iterators have [[FileStatus]]s for the same file path, keep the one from
+   * `iterWithPrecedence` and discard that from `iter`.
+   */
+  private def mergeFileIterators(
+    iter: Iterator[FileStatus],
+    iterWithPrecedence: Iterator[FileStatus]
+  ): Iterator[FileStatus] = {
+    val result = (iter.map(f => (f.getPath, f)).toMap
+      ++ iterWithPrecedence.map(f => (f.getPath, f))).values.toSeq
+      .sortBy(_.getPath.getName)
+    result.iterator
+  }
+
+  private def resolved(path: Path): (FileSystem, Path) = {
+    val fs = path.getFileSystem(getHadoopConfiguration)
+    val resolvedPath = stripUserInfo(fs.makeQualified(path))
+    (fs, resolvedPath)
+  }
+
+  private def stripUserInfo(path: Path): Path = {
+    val uri = path.toUri
+    val newUri = new URI(
+      uri.getScheme,
+      null,
+      uri.getHost,
+      uri.getPort,
+      uri.getPath,
+      uri.getQuery,
+      uri.getFragment
+    )
+    new Path(newUri)
+  }
```

Comment:
out of curiosity, why do we net to strip the user info from the `path`? I noticed the [`S3SingleDriverLogStore`](https://github.com/delta-io/delta/blob/ef76d4dc45748b2d3f7781555a740fe8954fde62/core/src/main/scala/org/apache/spark/sql/delta/storage/S3SingleDriverLogStore.scala#L55) also does it.

Reply:
I simply copied this code from S3SingleDriverLogStore - maybe S3 credentials may be passed this way?

Reply:
I see. But [`S3SingleDriverLogStore`](https://github.com/delta-io/delta/blob/ef76d4dc45748b2d3f7781555a740fe8954fde62/core/src/main/scala/org/apache/spark/sql/delta/storage/S3SingleDriverLogStore.scala#L55) seems to be removing the `userInfo` from the `URI`. This makes me think the credentials, if any, are removed. Not sure why. 

@tdas, do you happen to know?

Reply:
@scottsand-db what about this stripUserInfo? Is that needed?

Reply:
Doing what `S3SingledrigerLogStore` does seems fine with me.